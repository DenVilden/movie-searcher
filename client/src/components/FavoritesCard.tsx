import {
  CardActionArea,
  LinearProgress,
  Typography,
  CardMedia,
} from "@material-ui/core";
import { useRouter } from "next/router";
import styled from "styled-components";
import ErrorMessage from "../containers/ErrorMessage";
import { useGetMovieInfoQuery } from "../graphql";

const StyledTypography = styled(Typography)`
  padding: ${(props) => props.theme.spacing(2)}px;
`;

const CardWrapper = styled.div`
  display: flex;
  min-width: 200px;
`;

const StyledCardMedia = styled(CardMedia)`
  display: none;
  height: 56px;
  margin: ${(props) => props.theme.spacing(1)}px;
  width: 50px;

  ${(props) => props.theme.breakpoints.up("sm")} {
    display: block;
  }
`;

type Props = {
  toggleFavoritesOpen: () => void;
  id: string;
};

const FavoritesCard = ({ toggleFavoritesOpen, id }: Props) => {
  const { loading, error, data } = useGetMovieInfoQuery({ variables: { id } });

  const router = useRouter();

  if (error) return <ErrorMessage error={error} />;

  if (loading || !data?.movieInfo) return <LinearProgress />;

  return (
    <CardActionArea
      data-testid="favorites-card"
      onClick={() => {
        router.push(`/movie/${id}`);
        toggleFavoritesOpen();
      }}
    >
      <CardWrapper>
        <StyledCardMedia
          image={data.movieInfo.poster_path || "/no-image.jpg"}
          src="img"
          title={data.movieInfo.title}
        />
        <StyledTypography>{data.movieInfo.title}</StyledTypography>
      </CardWrapper>
    </CardActionArea>
  );
};

export default FavoritesCard;
