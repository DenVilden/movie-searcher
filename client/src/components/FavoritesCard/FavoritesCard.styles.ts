import { Typography } from "@material-ui/core";
import styled from "styled-components";

export const StyledTypography = styled(Typography)`
  padding: ${(props) => props.theme.spacing(3, 2, 2, 2)};
`;

export const CardWrapper = styled.div`
  display: flex;
  min-width: 200px;
`;

export const StyledCardMedia = styled("div")`
  display: none;
  height: 56px;
  position: relative;
  margin: ${(props) => props.theme.spacing(1)}px;
  width: 50px;

  ${(props) => props.theme.breakpoints.up("sm")} {
    display: block;
  }
`;
