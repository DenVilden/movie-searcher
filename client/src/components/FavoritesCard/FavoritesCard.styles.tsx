import { Typography, CardMedia } from '@material-ui/core';
import styled from 'styled-components';

export const StyledTypography = styled(Typography)`
  padding: ${props => props.theme.spacing(2)}px;
`;

export const CardWrapper = styled.div`
  display: flex;
  min-width: 200px;
`;

export const StyledCardMedia = styled(CardMedia)`
  display: none;
  height: 56px;
  margin: ${props => props.theme.spacing(1)}px;
  width: 50px;

  ${props => props.theme.breakpoints.up('sm')} {
    display: block;
  }
`;
