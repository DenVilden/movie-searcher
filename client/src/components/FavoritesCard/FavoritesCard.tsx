import React from 'react';
import { CardActionArea } from '@material-ui/core';
import { useHistory } from 'react-router-dom';
import noImage from '../../assets/no-image.jpg';
import {
  StyledTypography,
  CardWrapper,
  StyledCardMedia,
} from './FavoritesCard.styles';
import { MovieInfo } from '../../__generated__/types';

type Props = {
  toggleFavoritesOpen: () => void;
  movie: MovieInfo;
};

const FavoritesCard = ({ toggleFavoritesOpen, movie }: Props) => {
  const history = useHistory();

  return (
    <CardActionArea
      data-testid="favorites-card"
      onClick={() => {
        history.push(`/movie/${movie.id}`);
        toggleFavoritesOpen();
      }}
    >
      <CardWrapper>
        <StyledCardMedia
          image={movie.poster_path || noImage}
          src="img"
          title={movie.title}
        />
        <StyledTypography>{movie.title}</StyledTypography>
      </CardWrapper>
    </CardActionArea>
  );
};

export default FavoritesCard;
