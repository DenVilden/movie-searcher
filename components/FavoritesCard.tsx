import styled from '@emotion/styled'
import { MenuItem, Typography } from '@material-ui/core'
import Image from 'next/image'
import Link from 'next/link'

import { Favorite } from '@/apollo/client'

const Title = styled(Typography)`
  padding-left: 15px;
  white-space: normal;
  width: 100%;
`

interface Props {
  favorite: Favorite
  handleToggle: () => void
}

export default function FavoritesCard({
  favorite: { id, media_type, poster_path, title },
  handleToggle,
}: Props) {
  return (
    <Link href={`/${media_type}/${id}`}>
      <MenuItem onClick={handleToggle}>
        <Image
          alt={title}
          height={50}
          layout="fixed"
          objectFit="cover"
          src={poster_path || '/no-image.png'}
          width={50}
        />
        <Title>{title}</Title>
      </MenuItem>
    </Link>
  )
}
