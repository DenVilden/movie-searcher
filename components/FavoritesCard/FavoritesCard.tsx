import { CardActionArea, Typography } from '@material-ui/core';
import { useRouter } from 'next/router';
import Image from 'next/image';
import styled from '@emotion/styled';
import { Favorite } from '../../apollo';

const StyledTypography = styled(Typography)`
  padding: ${(props) => props.theme.spacing(3, 2, 2, 2)};
`;

const CardWrapper = styled.div`
  display: flex;
  min-width: 200px;
`;

const ImageWrapper = styled.div`
  display: none;
  height: 56px;
  margin: ${(props) => props.theme.spacing(1)};
  position: relative;
  width: 50px;

  ${(props) => props.theme.breakpoints.up('sm')} {
    display: block;
  }
`;

interface Props {
  handleToggle: () => void;
  favorite: Favorite;
}

export default function FavoritesCard({
  handleToggle,
  favorite: { media_type, id, title, poster_path },
}: Props) {
  const router = useRouter();

  return (
    <CardActionArea
      onClick={() => {
        router.push(`/${media_type}/${id}`);
        handleToggle();
      }}
    >
      <CardWrapper>
        <ImageWrapper>
          <Image
            layout="fill"
            alt={title}
            src={poster_path || '/no-image.jpg'}
          />
        </ImageWrapper>
        <StyledTypography>{title}</StyledTypography>
      </CardWrapper>
    </CardActionArea>
  );
}
