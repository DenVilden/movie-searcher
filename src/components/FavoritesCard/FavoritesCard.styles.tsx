import { Typography, CardMedia, Theme } from '@material-ui/core';
import styled from 'styled-components';

export const StyledTypography = styled(Typography)`
  padding: ${({ theme }: { theme: Theme }) => theme.spacing(2)}px;
`;

export const CardWrapper = styled.div`
  display: flex;
  min-width: 200px;
`;

export const StyledCardMedia = styled(CardMedia)`
  display: none;
  height: 56px;
  margin: ${({ theme }: { theme: Theme }) => theme.spacing(1)}px;
  width: 50px;

  ${({ theme }: { theme: Theme }) => theme.breakpoints.up('sm')} {
    display: block;
  }
`;
