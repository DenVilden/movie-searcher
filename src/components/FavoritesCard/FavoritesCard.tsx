import React from 'react';
import { CardActionArea } from '@material-ui/core';
import noImage from '../../assets/no-image.jpg';
import {
  StyledTypography,
  CardWrapper,
  StyledCardMedia,
} from './FavoritesCard.styles';

type Props = {
  clearInputValue: () => void;
  toggleFavoritesOpen: () => void;
  poster: string | null;
  title: string;
  goTo: () => void;
};

const FavoritesCard = ({
  clearInputValue,
  toggleFavoritesOpen,
  poster,
  title,
  goTo,
}: Props) => (
  <CardActionArea
    onClick={() => {
      goTo();
      clearInputValue();
      toggleFavoritesOpen();
    }}
  >
    <CardWrapper>
      <StyledCardMedia image={poster || noImage} src="img" title={title} />
      <StyledTypography>{title}</StyledTypography>
    </CardWrapper>
  </CardActionArea>
);

export default FavoritesCard;
