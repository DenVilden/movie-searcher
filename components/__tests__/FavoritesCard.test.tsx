import renderApollo from 'lib/setupTests';
import FavoritesCard from '../FavoritesCard';

const mock = {
  backdrop_path: null,
  budget: '0',
  id: 1,
  media_type: 'movie',
  overview: 'test data',
  poster_path: null,
  release_date: '2020',
  revenue: '0',
  similar: {
    results: [
      {
        id: 1,
        media_type: 'movie',
        poster_path: null,
        release_date: '2020',
        title: 'test',
      },
    ],
  },
  title: 'test',
  vote_average: 5,
};

describe('favoritesCard', () => {
  it('should take a snapshot', () => {
    const { asFragment } = renderApollo(
      <FavoritesCard favorite={mock} handleToggle={jest.fn()} />,
    );

    const element = asFragment();

    expect(element).toMatchSnapshot();
  });
});
