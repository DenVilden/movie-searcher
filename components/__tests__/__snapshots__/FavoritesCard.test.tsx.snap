// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`favoritesCard should take a snapshot 1`] = `
<DocumentFragment>
  <button
    class="MuiButtonBase-root MuiCardActionArea-root css-b6lsxj-MuiButtonBase-root-MuiCardActionArea-root"
    tabindex="0"
    type="button"
  >
    <div
      class="css-17b90ar-CardWrapper ev7japj1"
    >
      <div
        style="overflow: hidden; box-sizing: border-box; display: inline-block; position: relative; width: 50px; height: 50px;"
      >
        <img
          alt="title"
          decoding="async"
          src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
          style="visibility: hidden; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: cover;"
        />
      </div>
      <p
        class="MuiTypography-root MuiTypography-body1 ev7japj0 css-oyktpt-MuiTypography-root-StyledTypography"
      >
        title
      </p>
    </div>
    <span
      class="MuiCardActionArea-focusHighlight css-1v2exvi-MuiCardActionArea-focusHighlight"
    />
    <span
      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
    />
  </button>
</DocumentFragment>
`;
